Remove jupyterlab as a build dependency of jupyterlab_pygments to
avoid a dependency cycle. The jupyterlab dependency is not needed to
produce the same wheel content.

Refer to https://src.fedoraproject.org/rpms/python-jupyterlab_pygments/blob/rawhide/f/python-jupyterlab_pygments.spec#_53

diff -urNp jupyterlab_pygments-0.3.0/pyproject.toml.0 jupyterlab_pygments-0.3.0/pyproject.toml
--- jupyterlab_pygments-0.3.0/pyproject.toml.0	2024-05-17 18:42:42.651775817 -0400
+++ jupyterlab_pygments-0.3.0/pyproject.toml	2024-05-17 18:42:50.675704490 -0400
@@ -1,7 +1,6 @@
 [build-system]
 requires = [
     "hatchling>=1.5.0",
-    "jupyterlab>=4.0.0,<5",
     "hatch-nodejs-version>=0.3.2",
 ]
 build-backend = "hatchling.build"
